src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -306,7 +315,7 @@ private static boolean isSingleLineBlock(Details details) {
             nextToken = Details.getNextToken(doWhileSemi);
         }
         return rcurly.getLineNo() == lcurly.getLineNo()
-                && rcurly.getLineNo() != nextToken.getLineNo();
+                && (rcurly.getLineNo() != nextToken.getLineNo() || endsWithSemicolon(details));
```

Comment:
Maybe I am just not understanding why the semicolon plays a role in this logic.

Reply:
@rnveach The semicolon plays a role here because if the condition is `rcurly.getLineNo() != nextToken.getLineNo()` then for cases such as `public class TestClass {};` it will result in a violation because in this case the `nextToken` which is the semicolon is on the same line as `rcurly` thus to be able to account for cases where the `CLASS_DEF` or `METHOD_DEF` end with a semicolon `(rcurly.getLineNo() != nextToken.getLineNo() || endsWithSemicolon(details))` has to be used.

Reply:
Your reason just details alot of the code but I am questioning the higher level reasoning. The method is called `isSingleLine`. Semicolon has no bearing on if a block is a single line or not. So I don't understand why we are checking semicolon in this method.

Maybe the issue is the name of the method as it is already checking the next token. If this is so, maybe it should be renamed to something like `isSingleLineBlockAlone` and have the javadoc updated.

Please confirm.

Reply:
@rnveach Yes the method should be renamed to `isSingleLineBlockAlone` I will make the change.

Reply:
@rnveach I think it should be named as `isSingleLineBlockAndAloneOnLine` because it performs both the checks (i.e. Checking weather the block is on a single line and weather the block is alone on line)

Reply:
I have changes the method name to `isBlockAloneOnSingleLine` and have updated the javadoc accordingly if any change is required in this please let me know.